### PR TITLE
UCP: Add error traces for address pack/unpack errors

### PR DESCRIPTION
**Pull request by brminich**
## What
Add error traces for UCP address pack/unpack failures

## Why ?
Better debugging experience

### CODE REVIEW

**Review by hoopoepg — APPROVED** [review]

File: src/ucp/wireup/address.c [hunk 1]
```diff
@@ -1225,6 +1233,10 @@ ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep, void *buffer, size_t size,
             iface_attr = &wiface->attr;
 
             if (!ucp_worker_iface_can_connect(iface_attr)) {
+                if (!(pack_flags & UCP_ADDRESS_PACK_FLAG_NO_TRACE)) {
+                    ucs_error("failed to pack address, rsc[%d].cap.flags 0x%lx",
```
Comment (hoopoepg) [inline src/ucp/wireup/address.c:1237]: minor: may be `failed to pack address: iface can't connect, ....`

**Review by hoopoepg — APPROVED** [review]

File: src/ucp/wireup/address.c [hunk 2]
```diff
@@ -1212,6 +1216,10 @@ ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep, void *buffer, size_t size,
             status = uct_iface_get_device_address(wiface->iface,
                                                   (uct_device_addr_t*)ptr);
             if (status != UCS_OK) {
+                if (!(pack_flags & UCP_ADDRESS_PACK_FLAG_NO_TRACE)) {
+                    ucs_error("failed to get device address %s",
```
Comment (yosefe) [inline src/ucp/wireup/address.c:1220]: can we print device name?

File: src/ucp/wireup/address.c [hunk 3]
```diff
@@ -1225,6 +1233,10 @@ ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep, void *buffer, size_t size,
             iface_attr = &wiface->attr;
 
             if (!ucp_worker_iface_can_connect(iface_attr)) {
+                if (!(pack_flags & UCP_ADDRESS_PACK_FLAG_NO_TRACE)) {
+                    ucs_error("iface can't connect, rsc[%d].cap.flags 0x%lx",
```
Comment (yosefe) [inline src/ucp/wireup/address.c:1237]: can print more details?

File: src/ucp/wireup/address.c [hunk 4]
```diff
@@ -1261,6 +1273,10 @@ ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep, void *buffer, size_t size,
                 status = uct_iface_get_address(wiface->iface,
                                                (uct_iface_addr_t*)ptr);
                 if (status != UCS_OK) {
+                    if (!(pack_flags & UCP_ADDRESS_PACK_FLAG_NO_TRACE)) {
+                        ucs_error("failed to get iface address %s",
```
Comment (yosefe) [inline src/ucp/wireup/address.c:1277]: can print which transport/device?

File: src/ucp/wireup/address.c [hunk 5]
```diff
@@ -1292,6 +1308,10 @@ ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep, void *buffer, size_t size,
                     /* pack ep address */
                     status = uct_ep_get_address(ucp_ep_get_lane(ep, lane), ptr);
                     if (status != UCS_OK) {
+                        if (!(pack_flags & UCP_ADDRESS_PACK_FLAG_NO_TRACE)) {
+                            ucs_error("failed to get ep address %s",
```
Comment (yosefe) [inline src/ucp/wireup/address.c:1312]: can print which transport/device/lane?

File: src/ucp/wireup/address.c [hunk 6]
```diff
@@ -19,6 +19,16 @@
 #include <ucs/type/float8.h>
 #include <inttypes.h>
 
+#define ucp_address_error(_pack_flags, _fmt, ...) \
+    if (!(_pack_flags & UCP_ADDRESS_PACK_FLAG_NO_TRACE)) { \
```
Comment (yosefe) [inline src/ucp/wireup/address.c:23]: () around macro params

**Review by yosefe — DISMISSED** [review]
pls squash

**Review by yosefe — APPROVED** [review]

**Pull request merged**